gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/drilling/Drill.java

```diff
@@ -0,0 +1,251 @@
+import org.gradoop.flink.model.impl.operators.drilling.functions.drillfunctions.DrillFunction;
+
+import java.util.Objects;
+
+/**
+ * Base class for the drill up operation. The drill operator gives the opportunity to modify a
+ * property value of graph elements. This modification may be applied on all elements of a kind
+ * (vertex / edge) or only on elements with a specific label. The old value will be stored on the
+ * element to be able to restore the values later.
+ */
+public abstract class Drill implements UnaryGraphToGraphOperator {
+
+  /**
+   * Label of the element whose property shall be drilled.
+   */
+  private String label;
+  /**
+   * Property key.
+   */
+  private String propertyKey;
+  /**
+   * Drill function which shall be applied to a property.
+   */
+  private DrillFunction function;
+  /**
+   * New property key.
+   */
+  private String newPropertyKey;
+  /**
+   * True, if vertices shall be drilled, false for edges
+   */
+  private boolean drillVertex;
+  /**
+   * True, if all elements of a kind (vertex / edge) shall be drilled.
+   */
+  private boolean drillAllLabels;
+  /**
+   * True, if the current property key shall be reused.
+   */
+  private boolean keepCurrentPropertyKey;
+
+  /**
+   * Valued constructor.
+   *  @param label                  label of the element whose property shall be drilled
+   * @param propertyKey             property key
+   * @param function                drill function which shall be applied to a property
+   * @param newPropertyKey          new property key
+   * @param drillVertex             true, if vertices shall be drilled, false for edges
+   * @param drillAllLabels          true, if all elements of a kind (vertex / edge) shall be drilled
+   * @param keepCurrentPropertyKey  true, if the current property key shall be reused
+   */
+  Drill(String label, String propertyKey, DrillFunction function, String newPropertyKey,
+    boolean drillVertex, boolean drillAllLabels, boolean keepCurrentPropertyKey) {
+    this.label = label;
+    this.propertyKey = propertyKey;
+    this.function = function;
+    this.newPropertyKey = newPropertyKey;
+    this.drillVertex = drillVertex;
+    this.drillAllLabels = drillAllLabels;
+    this.keepCurrentPropertyKey = keepCurrentPropertyKey;
+  }
+
+  protected String getLabel() {
+    return label;
+  }
+
+  protected String getPropertyKey() {
+    return propertyKey;
+  }
+
+  protected DrillFunction getFunction() {
+    return function;
+  }
+
+  protected String getNewPropertyKey() {
+    return newPropertyKey;
+  }
+
+  /**
+   * True, if vertices shall be drilled, false for edges
+   *
+   * @return true for vertex drilling
+   */
+  protected boolean drillVertex() {
+    return drillVertex;
+  }
+
+  /**
+   * True, if all elements of a kind (vertex / edge) shall be drilled.
+   *
+   * @return true for drilling all elements
+   */
+  protected boolean drillAllLabels() {
+    return drillAllLabels;
```

Comment:
I think this is just `return label == null` so no need to specify this argument explicitly